### DISCUSSION TITLE
【Nav主题】把 slug 设置为超链接的 post 不显示

### DISCUSSION THREAD

**Discussion by W4ddl3** [tangly1024/NotionNext#2008] [Q&A] [answered]
**描述bug**
我把一个 post 的 slug 设置为一个实际存在的 url 后。
发现无论：
- 刷新自动抓取多少次
- 使用 yarn export 或者 默认的部署方式 重新部署无数次

它都不会显示这个 post ，只有把 slug 改成字符串才行

**复现步骤**
如描述所说
图1,2 为 slug 设置为 url 的结果（不显示）
图3,4 为 slug 设置为 字符串 的结果（正常显示）

很早的时候，NotionNext 还不太支持 slug 设置为一个 url （Vercel 部署会失败），后来 issue 里有个人提到了，您就修复了。
没修复这个 bug 前，我测试即使没法手动部署，但是借助 Vercel 自动抓取，仍然可以把一个 post 的 slug 设置为 url ，并且实现跳转这个 url 的功能。
我注意到这个 bug 修复后，就想试试，忽然发现了当前这个问题。

我是比较早就开始用 NotionNext ，我自己作为开发者也思考了原因，但是怎么排除都判断不出来是谁的问题。总不能是 Notion 的问题吧😂
我看到有同样用 Nav 主题的实现了这个功能，就怎么也想不通

**期望的正常结果**
正常显示

**截图**
![Screenshot_20240216_114227](https://github.com/tangly1024/NotionNext/assets/141616559/0024eef3-d4a6-48d8-8850-a6240fe3298d)
![Screenshot_20240216_114314](https://github.com/tangly1024/NotionNext/assets/141616559/39bf79c1-78ad-48ac-83ff-46a341d56dea)
![Screenshot_20240216_114334](https://github.com/tangly1024/NotionNext/assets/141616559/78ff6cb2-6853-4314-accc-4fb838715d5e)
![Screenshot_20240216_114423](https://github.com/tangly1024/NotionNext/assets/141616559/c57e8eab-42f1-48f7-a3c8-7fcb333f504d)

**环境**

- 【必填】NotionNext版本 [最新版]
- 【必填】主题 [nav]
- 【必填】部署方案 [vercel]

**Comment by tammyxx** [comment]
我也是nav主题，slug设置url 就是不会跳转

**Comment by bbb3n** [comment]
> 我也是nav主题，slug设置url 就是不会跳转

那你就是还能显示喽？
这个人的网站是能显示也能跳转
laogou717.com

**Comment by tangly1024** [comment]
路过，我刚在本地调试了这个功能， slug为超链接的文章 显示正常，跳转也正常，版本是最新的。

![image](https://github.com/tangly1024/NotionNext/assets/15920488/9d2176bb-fa66-4f5d-896c-bcca3d0f4e0b)

这是我的模板，似乎me没有特别的额外设置。

![image](https://github.com/tangly1024/NotionNext/assets/15920488/e781f2dc-8479-442b-bac7-da4d6452ac30)

要怎么做才能复现这个bug呢？

**Comment by W4ddl3** [comment]
> 路过，我刚在本地调试了这个功能， slug为超链接的文章 显示正常，跳转也正常，版本是最新的。
> 
> ![image](https://private-user-images.githubusercontent.com/15920488/305683335-9d2176bb-fa66-4f5d-896c-bcca3d0f4e0b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgyMzEzNjksIm5iZiI6MTcwODIzMTA2OSwicGF0aCI6Ii8xNTkyMDQ4OC8zMDU2ODMzMzUtOWQyMTc2YmItZmE2Ni00ZjVkLTg5NmMtYmNjYTNkMGY0ZTBiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE4VDA0Mzc0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMxY2M1ODBkMDY3ZWMyNjI5ZjVhNzcwYWIxMmNhZmI5OTMyMTA5MTljNWE3MGFlMzhiMjUwYmRlODJkZDRkMzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.m34YYGZ_a5iJOBntNJ2adGoVE3PORnGIssWFgr_WbWs)
> 
> 这是我的模板，似乎me没有特别的额外设置。
> 
> ![image](https://private-user-images.githubusercontent.com/15920488/305683346-e781f2dc-8479-442b-bac7-da4d6452ac30.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgyMzEzNjksIm5iZiI6MTcwODIzMTA2OSwicGF0aCI6Ii8xNTkyMDQ4OC8zMDU2ODMzNDYtZTc4MWYyZGMtODQ3OS00NDJiLWJhYzctZGE0ZDY0NTJhYzMwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE4VDA0Mzc0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdkYzA5ZGY2M2M4YjdmNGJiYzc3NTNlMTViNWY5ZjBkYTc2ODkyY2ZiYjQ5ZjNlODEzYThiOGJkYTA2MTQzNmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.YqMuIjb-D9D40JU0ZxGaL_ZGKKFJzvqO1Wq6D9XxF7s)
> 
> 要怎么做才能复现这个bug呢？



破案了，破案了

我最开始也是百思不得其解
- 我使用全新模板和项目 部署 ，发现没问题
- 判断那肯定是文章模板的问题
- 排查了 各种属性的排列顺序（status tag category 这种），我以为我之前改了顺序导致什么错乱了。发现无变化
- 排查了 文章 config 功能（在 Notion 文章里直接配置的），因为我很早就用，所以没这个，后来也没加。发现无变化

- 然后我又回头看了看别人成功的 blog ，再想想还有什么变量被我改了
- 忽然我看到地址栏，别人的是 example.com/post/123 或者是 example.com/article/123 ，而我的是 example.com/123
- 遂即开始测试

图1是我啥都配置没改，只是把 5 号文章 slug 设置为链接
图2是我把 那个 `NEXT_PUBLIC_POST_URL_PREFIX` 设置为 空，其他和上面一样
**这时你会发现，5 号 文章消失了家人们**

![Screenshot_20240218_122619](https://github.com/tangly1024/NotionNext/assets/141616559/62e26d2a-1b5c-47ed-a4cd-0eee3f945653)
![Screenshot_20240218_123727](https://github.com/tangly1024/NotionNext/assets/141616559/89eb440b-49aa-4ca5-a579-3ff8b5d25647)

这个 BUG 困扰我好久我测，哎

**Comment by W4ddl3** [comment, marked as answer]
总结，问题在于 `NEXT_PUBLIC_POST_URL_PREFIX` 这个变量为 空 值时导致的异常

另外我发现：
我要把这个设置为 空 的话，只能在 代码 里改成 '' 
没法在 Vercel 里设置这个 空 的环境变量。

或许这就是 bug 所在？

**Comment by tammyxx** [comment]
我也解决了，谢谢～

**Comment by tangly1024** [comment]
@W4ddl3 感谢大侦探 ， 确认是这个问题

> NEXT_PUBLIC_POST_URL_PREFIX 设置为空 ， 将导致Slug为超链接的Post丢失
